File: makeChapters.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
)

type ChapterData struct {
	// Define your JSON structure here
	// Adjust fields based on actual JSON structure
	Chapter string `json:"chapter"`
	// Add other relevant fields
}

func main() {
	sourceDir := `D:\Projects\oper\MBh`
	outputDir := filepath.Join(sourceDir, "processed_chapters")

	// Create output directory if not exists
	os.MkdirAll(outputDir, 0755)

	// Organized chapters map
	chapterGroups := make(map[string][]ChapterData)

	// Iterate through MBh01.json to MBh18.json
	for i := 1; i <= 18; i++ {
		filename := fmt.Sprintf("MBh%02d.json", i)
		filePath := filepath.Join(sourceDir, filename)

		// Read JSON file
		data, err := ioutil.ReadFile(filePath)
		if err != nil {
			fmt.Printf("Error reading %s: %v\n", filename, err)
			continue
		}

		var chapters []ChapterData
		err = json.Unmarshal(data, &chapters)
		if err != nil {
			fmt.Printf("Error parsing %s: %v\n", filename, err)
			continue
		}

		// Determ

		// Group chapters
		for _, chapter := range chapters {
			chapterKey := extractChapterKey(chapter.Chapter)
			chapterGroups[chapterKey] = append(chapterGroups[chapterKey], chapter)
		}

		// Create book-specific directory
		bookDir := filepath.Join(outputDir, fmt.Sprintf("Book_%02d", i))
		os.MkdirAll(bookDir, 0755)

		// Write chapter files
		for chapterKey, chapterData := range chapterGroups {
			chapterFile := filepath.Join(bookDir, chapterKey+".json")
			chapterJSON, _ := json.MarshalIndent(chapterData, "", "  ")
			ioutil.WriteFile(chapterFile, chapterJSON, 0644)
		}

		// Clear for next iteration
		chapterGroups = make(map[string][]ChapterData)
	}
}

func extractChapterKey(chapter string) string {
	// Extract chapter number from key format
	if len(chapter) >= 8 {
		return chapter[2:8]
	}
	return chapter
}

File: makeChapters.py
import os
import json
import re

def extract_chapter_number(key):
    """Extract chapter number from the key format."""
    match = re.match(r'(\d{2})001(\d{4})', key)
    return match.group(0) if match else key

def process_mbh_files():
    source_dir = r'D:\Projects\oper\MBh'
    output_dir = os.path.join(source_dir, 'processed_chapters')
    os.makedirs(output_dir, exist_ok=True)

    # Process each MBh JSON file
    for book_num in range(1, 19):
        filename = f'MBh{book_num:02d}.json'
        filepath = os.path.join(source_dir, filename)
        
        # Skip if file doesn't exist
        if not os.path.exists(filepath):
            print(f"File {filename} not found. Skipping.")
            continue

        # Read JSON file
        with open(filepath, 'r', encoding='utf-8') as f:
            data = json.load(f)

        # Group chapters by chapter number
        chapter_groups = {}
        for key, value in data.items():
            chapter_num = extract_chapter_number(key)
            if chapter_num not in chapter_groups:
                chapter_groups[chapter_num] = {}
            chapter_groups[chapter_num][key] = value

        # Create book-specific directory
        book_dir = os.path.join(output_dir, f'Book_{book_num:02d}')
        os.makedirs(book_dir, exist_ok=True)

        # Write chapter files
        for chapter_num, chapter_data in chapter_groups.items():
            chapter_file = os.path.join(book_dir, f'{chapter_num}.json')
            with open(chapter_file, 'w', encoding='utf-8') as f:
                json.dump(chapter_data, f, ensure_ascii=False, indent=2)

        print(f"Processed {filename}: {len(chapter_groups)} chapters")

if __name__ == '__main__':
    process_mbh_files()
